mne/io/edf/edf.py

```diff
@@ -565,7 +602,7 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, tal_channel,
     # sfreq defined as the max sampling rate of eeg
     picks = pick_types(info, meg=False, eeg=True)
     info['sfreq'] = n_samps[picks].max() / float(record_length)
-    edf_info['nsamples'] = int(n_records * info['sfreq'])
+    edf_info['nsamples'] = int(n_records * info['sfreq'] * record_length)
```

Comment:
are `n_records` and `record_length` both ints? To avoid rounding errors with big float multiplications, it's probably safer to do this instead:
```
n_records * int(info['sfreq']) * record_length
```